From 497ea0086817d3443253457d94b73bf2f190b810 Mon Sep 17 00:00:00 2001
From: Enrico Berti <hello@enricoberti.com>
Date: Fri, 31 Oct 2014 11:06:36 +0100
Subject: [PATCH 0085/1173] HUE-2430 [pig] Progress bars of running scripts
 not updated on Dashboard

Changed dashboard to be refreshed also when scripts are not running
Fixed a but with failed logs retrieval
---
 apps/pig/src/pig/api.py             |    1 +
 apps/pig/src/pig/templates/app.mako |   11 +----------
 apps/pig/static/js/pig.ko.js        |    1 -
 3 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/apps/pig/src/pig/api.py b/apps/pig/src/pig/api.py
index c28fb2d..4f0d93d 100644
--- a/apps/pig/src/pig/api.py
+++ b/apps/pig/src/pig/api.py
@@ -159,6 +159,7 @@ class OozieApi:
 
       except Exception, e:
         LOG.error('An error happen while watching the demo running: %(error)s' % {'error': e})
+        is_really_done = True
 
     workflow_actions = []
 
diff --git a/apps/pig/src/pig/templates/app.mako b/apps/pig/src/pig/templates/app.mako
index 9b4aa88..26ae5bf 100644
--- a/apps/pig/src/pig/templates/app.mako
+++ b/apps/pig/src/pig/templates/app.mako
@@ -1115,10 +1115,6 @@ ${ commonshare() | n,unicode }
       showAlert("<b>" + viewModel.currentScript().name() + "</b> ${_('has been saved correctly.')}");
     });
 
-    $(document).on("refreshDashboard", function () {
-      refreshDashboard();
-    });
-
     $(document).on("showDashboard", function () {
       routie("dashboard");
       $.jHueTitleUpdater.reset();
@@ -1168,15 +1164,10 @@ ${ commonshare() | n,unicode }
 
     refreshDashboard();
 
-    var dashboardRefreshInterval = window.setInterval(function () {
-      if (viewModel.runningScripts().length > 0) {
-        refreshDashboard();
-      }
-    }, 3000);
-
     function refreshDashboard() {
       $.getJSON("${ url('pig:dashboard') }", function (data) {
         viewModel.updateDashboard(data);
+        window.setTimeout(refreshDashboard, viewModel.runningScripts().length > 0 ? 3000 : 10000);
       });
     }
 
diff --git a/apps/pig/static/js/pig.ko.js b/apps/pig/static/js/pig.ko.js
index 8ee7930..6768d63 100644
--- a/apps/pig/static/js/pig.ko.js
+++ b/apps/pig/static/js/pig.ko.js
@@ -542,7 +542,6 @@ var PigViewModel = function (props) {
           script.isRunning(true);
           script.watchUrl(data.watchUrl);
           $(document).trigger("startLogsRefresh");
-          $(document).trigger("refreshDashboard");
           self.updateScripts();
         }, "json").fail( function(xhr, textStatus, errorThrown) {
           $(document).trigger("error", xhr.responseText);
-- 
1.7.9.5

